### PR TITLE
Fixed Wrong Explanation of Function Expression Hoisting

### PR DESCRIPTION
**Pull request by ParasRawat29**
#### Summary

Fixed Wrong explanation of Function Expression Hoisting .
It was not accurate . It is stated that "Function Expression are not Hoisted" but that's not true they are hoisted and intially initialized as undefined.
#### Motivation

My change will help the readers to know the actual reason behind the concept of function expression hoisting making their concept clear.
#### Supporting details

https://www.youtube.com/watch?v=SHINoHxvTso&ab_channel=AkshaySaini
timestamp : 04:04
#### Metadata

- [ ] Adds a new document
- [ x] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

**Comment by sideshowbarker** [comment]
As far as I can see, the existing MDN article is correct as written, and the change in this PR would make it incorrect.

If you have an actual test case that demonstrates a function expression getting hoisted, please add it in a comment here. Then this PR could be revisited and possibly re-opened.

**Comment by sideshowbarker** [comment]
To be a bit more specific: What’s immediately incorrect about the change in this PR is the first 7 words: *“When function expressions in javascript are hoisted”*.

The part of the change which follows those 7 words is closer to describing what actually happens. But what’s in that description is closer to describing what actually happens. But what actually happens doesn’t amount to the function expression itself being hoisted.

One problem here is that “hoisting” is an imprecise term that’s not normatively defined in the ES spec. So *“Function expressions in JavaScript are not hoisted”* is a common shorthand for describing something that, to be described more precisely, requires quite a bit more words to explain.

https://stackoverflow.com/questions/3887408/javascript-function-declaration-and-evaluation-order/3887590#3887590 has one of the best explanations I’ve come across. In particular, it includes this paragraph:

> Expressions unlike declarations are processed in the execution phase rather than the compilation phase. And because of this the order of expressions matter.

So while because of the term *“hoisting”* being an imprecise term, it’s may not seem inaccurate to talk about in terms of the _variable_ getting hoisted, it is just not correct to state that overall function expression gets hoisted.

The bottom line is that the existing statement *“Function expressions in JavaScript are not hoisted”* in MDN is consistent with statements made in many other JavaScript guides and tutorials. And while it’s true that if we were to add a much-longer explanation to the MDN article (similar to the explanation in the Stack Overflow answer I cited), it would be more precise than talking about “hoisting”. But the effect for developers would basically not change; what I mean is, the existing statement *“Function expressions in JavaScript are not hoisted”* achieves the goal of alerting developers to that fact that if you try to use a function expression out of execution order (in the way you can with a function declaration), it’s not going to work the way you’d expect — and so, don’t do it.
